src/tui/mainview.zig

```diff
@@ -550,6 +550,43 @@ const cmds = struct {
     }
     pub const add_diagnostic_meta = .{ .arguments = &.{ .string, .string, .string, .string, .integer, .integer, .integer, .integer, .integer } };
 
+    pub fn rename_symbol_item(self: *Self, ctx: Ctx) Result {
+        // because the incoming message is an array of Renames, we manuallly
+        // parse instead of using ctx.args.match() which doesn't seem to return
+        // the parsed length needed to correctly advance iter.
+        var iter = ctx.args.buf;
+        var len = try cbor.decodeArrayHeader(&iter);
+        var mroot: ?@import("Buffer").Root = null;
+        while (len != 0) {
+            len -= 1;
+            std.debug.assert(try cbor.decodeArrayHeader(&iter) == 6);
+            var file_uri: []const u8 = undefined;
+            if (!try cbor.matchString(&iter, &file_uri)) return error.MissingArgument;
+            var sel: ed.Selection = .{};
+            if (!try cbor.matchInt(usize, &iter, &sel.begin.row)) return error.MissingArgument;
+            if (!try cbor.matchInt(usize, &iter, &sel.begin.col)) return error.MissingArgument;
+            if (!try cbor.matchInt(usize, &iter, &sel.end.row)) return error.MissingArgument;
+            if (!try cbor.matchInt(usize, &iter, &sel.end.col)) return error.MissingArgument;
+            var new_text: []const u8 = undefined;
+            if (!try cbor.matchString(&iter, &new_text)) return error.MissingArgument;
+
+            if (self.get_active_editor()) |editor| {
+                // TODO match file_uri correctly. endsWith() isn't correct because 'path' is a
+                // short, relative path while 'file_uri' is an absolute path starting with 'file://'
+                const match = if (editor.file_path) |path| std.mem.endsWith(u8, file_uri, path) else false;
```

Comment:
This is the first TODO

Reply:
This should be done the same way as in `add_diagnostics`. ie:

```
file_path = project_manager.normalize_file_path(file_path);
if (self.get_active_editor()) |editor| if (std.mem.eql(u8, file_path, editor.file_path orelse "")) {
    ...
}
```

Reply:
done